### PR TITLE
DSND-1536 Implement new ApiService to call resource-changed endpoint

### PR DESCRIPTION
**Pull request by kinpang-CH**
* Implement ResourceChangedRequestMapper and models
* Add unit tests

Resolves [DSND-1536](https://companieshouse.atlassian.net/browse/DSND-1536)

[DSND-1536]: https://companieshouse.atlassian.net/browse/DSND-1536?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/company_appointments/mapper/ResourceChangedRequestMapper.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    private final Supplier<String> timestampGenerator;
+
+    public ResourceChangedRequestMapper(Supplier<String> timestampGenerator) {
+        this.timestampGenerator = timestampGenerator;
+    }
+
+    public ChangedResource mapChangedResource(ResourceChangedRequest request) {
+        ChangedResourceEvent event = new ChangedResourceEvent().publishedAt(this.timestampGenerator.get());
+        ChangedResource changedResource = new ChangedResource()
+                .resourceUri(String.format("company/%s/officers", request.getCompanyNumber()))
```
Comment (cgregoryCH) [inline src/main/java/uk/gov/companieshouse/company_appointments/mapper/ResourceChangedRequestMapper.java:21]: the resource URI should contain the appointmentId for the appointment which has changed

**Review by cgregoryCH — COMMENTED** [review]
The ResourceChangedRequestMapperTest has some declarations which can be made private. It is also possible to give the tests display names using the `Named.of()` method from JUnit5. See https://github.com/companieshouse/company-appointments.api.ch.gov.uk/pull/82/files#diff-d998410164fd6496f43861ac71251c349d8df691131e50402955c06e1e318acdR50 for an example.

**Review by johncook-ch — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 2]
```diff
@@ -16,7 +16,7 @@
     <properties>
         <maven.compiler.source>1.8</maven.compiler.source>
         <maven.compiler.target>1.8</maven.compiler.target>
-        <spring-boot-dependencies.version>2.5.6</spring-boot-dependencies.version>
+        <spring-boot-dependencies.version>2.7.9</spring-boot-dependencies.version>
```
[the remark this answers is not in the document]
Comment (johncook-ch) [reply pom.xml:19]: Also the parent POM version should be incremented to `1.3.1`

**Comment by johncook-ch** [comment]
When I ru `make clean test` the integration tests are failing with 
```
Error starting ApplicationContext. To display the conditions report re-run your application with 'debug' enabled.
```
The `master` branch builds fine

File: src/main/java/uk/gov/companieshouse/company_appointments/mapper/ResourceChangedRequestMapper.java (hunk 1 above)
Comment (kinpang-CH) [reply src/main/java/uk/gov/companieshouse/company_appointments/mapper/ResourceChangedRequestMapper.java]: Resolved in commit: https://github.com/companieshouse/company-appointments.api.ch.gov.uk/pull/80/commits/e2635f2cbba4f7401f8de2a618f1b5c29b21ed73

File: pom.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (kinpang-CH) [reply pom.xml]: Changes made in a seperate PR, merged in main and this is resolved now.

**Review by sthompsonCH — APPROVED** [review]

**Pull request closed without merging**